remote-workstations/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -243,131 +238,326 @@ active_d = ldap_lib(args.address, args.user_name, args.password, args.domain)
+        (sudo crontab -l 2>/dev/null; echo "@reboot bash $script_file $AGENT_TYPE") | sudo crontab -
+
+        #exit and restart VM
+        exit_restart
+    fi	
+    
+    if [ "$INST_LAST_STEP" == "step4 done" ]
+    then
+        echo "step5 starting" | tee -a $INST_LOG_FILE
+
+        install_lis
+    
+        EXIT_CODE=$?
+    
+        if [ $EXIT_CODE -eq 0 ]
+        then 
+            INST_LAST_STEP="step5 done"
+        else
+            INST_LAST_STEP="step5 failure: $EXIT_CODE"
+        fi
+
+        echo "$INST_LAST_STEP" | tee -a $INST_LOG_FILE
+    
+        if [ $EXIT_CODE -eq 0 ]
+        then	
+            exit_restart
         fi
-        sleep 60
     fi
-done
-
-# Install and setup the Sumo Collector
-echo "-->Install SumoLogic collector"
-mkdir /tmp/sumo
-wget "https://collectors.sumologic.com/rest/download/linux/64" -O /tmp/sumo/SumoCollector.sh && sudo chmod +x /tmp/sumo/SumoCollector.sh
-wget "$STORAGEURI/user.properties$SAS_TOKEN" -O /tmp/sumo/user.properties
-wget "$STORAGEURI/sumo-agent-vm-linux.json$SAS_TOKEN" -O /tmp/sumo/sumo-agent-vm-linux.json
-JSON_FILE=/tmp/sumo/sumo-agent-vm-linux.json
-echo "Attemtping to set sumo collector ID to: " "$COLLECTOR_ID"
-sed -i s/collectorID/"$COLLECTOR_ID"/ /tmp/sumo/user.properties
-sed -i 's|syncsourceFile|'$JSON_FILE'|' /tmp/sumo/user.properties
-
-sudo /tmp/sumo/SumoCollector.sh -q -varfile user.properties
-
-# service collector install - configures the collector to start at boot time
-service collector install
-service collector restart
-service collector status
-
-# Install idle shutdown script
-echo "-->Install idle shutdown"
-mkdir /tmp/idleShutdown
-wget "$STORAGEURI/Install-Idle-Shutdown.sh$SAS_TOKEN" -O /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh 
-awk '{ sub("\r$", ""); print }' /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh > /tmp/idleShutdown/Install-Idle-Shutdown.sh && sudo chmod +x /tmp/idleShutdown/Install-Idle-Shutdown.sh
-sudo /tmp/idleShutdown/Install-Idle-Shutdown.sh -install
-
-# Install Desktop
-echo "-->Install desktop"
-# sudo yum -y groupinstall "Server with GUI"
-sudo yum -y groupinstall 'X Window System' 'GNOME'
-
-# install firefox
-echo "-->Install firefox"
-sudo yum -y install firefox
-
-echo "-->set default graphical target"
-# The below command will change runlevel from runlevel 3 to runelevel 5 
-sudo systemctl set-default graphical.target
-
-echo "-->start graphical target"
-sudo systemctl start graphical.target
-
-(sleep 2;  sudo shutdown -f -r +0)&
-exit 0
\ No newline at end of file
+
+    if [ "$INST_LAST_STEP" == "step5 done" ]
+    then
+        echo "step6 starting" | tee -a $INST_LOG_FILE
+
+        install_nvidia_driver
+        EXIT_CODE=$?
+    
+        if [ $EXIT_CODE -eq 0 ]
+        then 
+            INST_LAST_STEP="step6 done"
+        else
+            INST_LAST_STEP="step6 failure: $EXIT_CODE"
+        fi
+
+        echo "$INST_LAST_STEP" | tee -a $INST_LOG_FILE
+    fi
+fi
+
+#remove job
+sudo crontab -r
```

Comment:
Does running this of a crontab job was never created do anything? It looks like it's possible to get here without creating cronjob

Reply:
yes for standard agent, there is no cronjob.  and remove operation is harmless operation here

Reply:
if fact, the can move up into if statement